src/main.ts

```diff
@@ -123,39 +123,55 @@ function loadTscConfig(args: string[], allDiagnostics: ts.Diagnostic[]):
   return {options, fileNames};
 }
 
+export interface ClosureJSOptions {
+  tsickleCompilerHostOptions: tsickle.Options;
+  tsickleHost: tsickle.TsickleHost;
+  files: Map<string, string>;
+  tsicklePasses: tsickle.Pass[];
+}
+
+function getDefaultClosureJSOptions(fileNames: string[], settings: Settings): ClosureJSOptions {
+  return {
+    tsickleCompilerHostOptions: {
+      googmodule: true,
+      es5Mode: false,
+      untyped: settings.isUntyped,
+    },
+    tsickleHost: {
+      shouldSkipTsickleProcessing: (fileName) => fileNames.indexOf(fileName) === -1,
+      pathToModuleName: cliSupport.pathToModuleName,
+      shouldIgnoreWarningsForPath: (filePath) => false,
+      fileNameToModuleId: (fileName) => fileName,
+    },
+    files: new Map<string, string>(),
+    tsicklePasses: [tsickle.Pass.CLOSURIZE],
+  };
+}
+
 /**
  * Compiles TypeScript code into Closure-compiler-ready JS.
  * Doesn't write any files to disk; all JS content is returned in a map.
  */
 export function toClosureJS(
     options: ts.CompilerOptions, fileNames: string[], settings: Settings,
-    allDiagnostics: ts.Diagnostic[],
-    files = new Map<string, string>()): {jsFiles: Map<string, string>, externs: string}|null {
+    allDiagnostics: ts.Diagnostic[], closureJSOPtions = {} as Partial<ClosureJSOptions>):
+    {jsFiles: Map<string, string>, externs: string}|null {
+  const fullClosureJSOptions: ClosureJSOptions = {
```

Comment:
I don't like the name "full" here because it's meaningless in the rest of the function.
What do you think?  Maybe you have a better idea?